---
title: Lints
description: All the lints provided by solidart_lint
---

# Lints

### avoid_dynamic_provider

`Provider` cannot be dynamic

**Bad**:

```dart
Solid(
  providers: [
    Provider(create: () => MyClass()),
  ],
),
```

**Good**:

```dart
Solid(
  providers: [
    Provider<MyClass>(create: () => MyClass()),
  ],
),
```

### invalid_solid_get_type

Specify the provider or signal type you want to get.

**Bad**:

```dart
final provider = context.get();
```

**Good**:

```dart
final provider = context.get<MyClass>();
```

### invalid_update_type

The update type is invalid, must not implement `SignalBase`.
You cannot update a signal that implements `SignalBase`, like `Signal`, `ReadSignal` and `Resource`.

**Bad**:

```dart
context.update<Signal<int>>('counter', (value) => value * 2);
```

**Good**:

```dart
context.update<int>('counter', (value) => value * 2);
```
